Admin

chasew2 opened the pull request:
Deleted instances folder to change the user table 
- added isAdmin boolean

Features added:
- Passcode page: passcode is 12, this leads to admin register page
- Admin register: Creates a user like normal but changes isAdmin boolean to true
- Admin login: Admins login like every user but are redirected to an admin dashboard on login.

What this has:
- Admin users are now added
- Admin dashboard page is started with the template of changing data 
- Admin auth.py code has 85% coverage testing 

What is doesn't have:
- Does not actually have code to update database yet, therefore the admin dashboard is incomplete (this will be completed later on, currently finishing connecting userId to cityId)
- Does not have tests for admin.py as that is incomplete code

iansteyn reviewed (changes requested):
# Ian's Review (1)
## ✅ What Works
- Able to access admin registration page
- Able to register a new admin account
- Able to login with admin account via regular login page
- Able to logout as admin
- Can view rest of website as normal while logged in as admin

## ⚠️ General issues
#### Admin.py
- obviously this is incomplete, but when you get around to it, make sure to separate it out into smaller functions

### New pages
- There appear to be two nearly identical files called `admin.html.jinja` and `admin_register.html.jinja`. Are they both supposed to be there?
- `admin_login.html.jinja` seems to be not currently used

### Jinja template blocks are incorrectly used.
Jinja blocks are not HTML tags. Instead, they mark sections of text that will be placed in a specific spot in the final HTML document, based on where those blocks are in the parent template. In particular:
- `{% block title %}`
  - Should specify the text that appears in the browser tab
  - Often does contain the same text as header
- `{% block header %}`
  - should not surround the title block (see the base template)
  - should specify the actual page header that appears beneath the navbar
  - should not be confused with `{% block head %}`, which is for including extra stylesheet tags or other stuff that goes in the `<head>` HTML section (on top of what is already in that tag in `base.html.jinja`)
- General resources
  - [Jinja Docs: Template Inheritance](https://jinja.palletsprojects.com/en/3.1.x/templates/#template-inheritance)
  - Look at how other templates in the project are written

 <hr>

## ⚠️🆕 Integrate new front end changes from main
Although there are no actual git 'conflicts' with this merge, some of the code/patterns don't line up with the new front end structure. I suggest you merge `main` into this branch and rework it to match up. You can read the pull request comments on #156 to learn more about these changes, and I would be happy to help/answer questions if you need.

#### Selector macros
- Use a `<table class="select-table">` element with predefined selector "macros" (bits of jinja that kind of work like functions). 
- See `main` `weather_summary.html` for an example of how this works.

#### JavaScript
- Move your JavaScript stuff to its own file, and write unit tests for that file. 
- Re-use script functions if you can (there is some helpful stuff in `submit.js` already, maybe you can add to that)
- Move any `<script>` tags to the new `{% block scripts %}`, below `{% block content %}`

chasew2 commented:
Updated admin with changes requested 
- Removed unused htmls 
- fixed blocks in htmls
- Moved starter js admin code into separate file
- merged with main
